### PR TITLE
Automatically generate OSG IID on the backend

### PR DESCRIPTION
**Pull request by mwestphall**
- Automatically generate OSG IID on `POST /institutions`
- Remove frontend components for manually inputting an OSG IID
- Validate ROR ID with a HEAD request to ror.org
- Reactivate deactivated institutions when recreated with the same name

### CODE REVIEW

**Comment by mwestphall** [comment]
This is already deployed on the test site, should be able to give it a quick look-over there

File: institutions-api/db/db.py [hunk 1]
```diff
@@ -30,6 +30,20 @@ def _full_osg_id(short_id):
...
+    ID_LENGTH = 12
+
+    # TODO actually guaranteeing uniqueness here might be a bit overkill based on ID length
+    all_ids = set(session.scalars(select(Institution.topology_identifier)).all())
+    for _ in range(MAX_TRIES):
+        short_id = ''.join([choice(ascii_lowercase + digits) for _ in range(ID_LENGTH)])
+        next_id = f"{OSG_ID_PREFIX}{short_id}"
+        if not next_id in all_ids:
+            return next_id
+        raise HTTPException(500, "Unable to generate new unique ID")
```
Comment (matyasselmeci) [inline institutions-api/db/db.py]:
```suggestion
    raise HTTPException(500, "Unable to generate new unique ID")
```
I think you meant this to be after the for loop

**Review by matyasselmeci — APPROVED** [review]

File: institutions-api/util/ror_utils.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from typing import Optional
+from fastapi import HTTPException
+import requests
+from ..models.api_models import ROR_ID_PREFIX
+
+
+def validate_ror_id(ror_id: Optional[str]):
+    """ Check whether an ROR ID is valid by performing a HEAD request to ror.org """
+    # TODO we might want to rate-limit this in some way
+    if ror_id and requests.head(ror_id).status_code != 200:
```
Comment (matyasselmeci) [inline institutions-api/util/ror_utils.py]:
```suggestion
    if ror_id and requests.head(ror_id, allow_redirects=True).status_code != 200:
```
According to https://requests.readthedocs.io/en/latest/user/quickstart/#redirection-and-history, HEAD does not follow redirects by default.

**Review by matyasselmeci — APPROVED** [review]
One change; pre-approving

**Pull request merged**